File: Nov2020/30Nov/FactorialGame.cpp
/*
Problem Name: Factorial Game
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/factorial-game-1/
*/

#include<bits/stdc++.h>
using namespace std;

#define test(t)     int t; cin>>t; while(t--)

int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);       
    long int fact[100001]={1}, mod = 1000000007; 
    for(int i=1;i<100001;++i){
        fact[i]=(i*fact[i-1])%mod;
    }    
    test(t){
        int n;
        cin>>n;
        cout<<fact[n]<<"\n";

    }
    return 0;
}

File: Nov2020/30Nov/FactorialGame.java
/*
Problem Name: Factorial Game
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/factorial-game-1/
*/
import java.io.*;
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        long m = 1000000007;
        long fact[] = new long[100000+1];
        fact[0] = 1;
        for(int i=1;i<=100000;i++)
            fact[i] = (i*fact[i-1])%m;
        int q = in.nextInt();
        for(;q>0;q--){
            out.println(fact[in.nextInt()]);
        }
        out.flush();
    }
}
